ethier/sign.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/divergencetech/ethier/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/spf13/cobra"
+)
+
+var signer *eth.Signer
+
+func init() {
+	var signCmd = &cobra.Command{
+		Use:   "sign",
+		Short: "Signs messages from stdin using an ECDSA signer.",
+	}
+
+	rootCmd.AddCommand(signCmd)
+	signCmd.PersistentFlags().String("mnemonic", "", "Path to file containing the mnemonic, password and account number (newline separated) for key derivation")
+
+	var signAddrCmd = &cobra.Command{
+		Use:     "addresses",
+		Short:   "Signs addresses from stdin using an ECDSA signer.",
+		PreRunE: getSigner,
+		RunE:    signAddresses,
+	}
+
+	signCmd.AddCommand(signAddrCmd)
+}
+
+type SignedAddress struct {
+	Address   string `json:"address"`
+	Signature string `json:"signature"`
+}
+
+func getSignerFromMnemonic(filepath string) (*eth.Signer, error) {
```

Comment:
I don't think we should be encouraging storage of mnemonics in plaintext, and adding this functionality does exactly that. I've got plans for an `ethier vault` command that will provide secure storage alternatives, at which point we can extend `ethier sign` to use it.

Reply:
I totally agree. This was only intended as an MVP until we switch to the `ethier vault` solution. The code has been removed.